src/app-layer-smtp.c

```diff
@@ -1139,6 +1139,15 @@ static int SMTPProcessRequest(SMTPState *state, Flow *f, AppLayerParserState *ps
     SCEnter();
     SMTPTransaction *tx = state->curr_tx;
 
+    state->toserver_data_count += (line->len + line->delim_len);
```

Comment:
Since this gets updated with line->len and delim_len, it was probably ok at its previous place as when len and delim_len are 0, nothing is updated here anyway.